pageserver/src/tenant/timeline/walreceiver.rs

```diff
@@ -63,6 +65,12 @@ pub struct WalReceiver {
     timeline_ref: Weak<Timeline>,
     conf: WalReceiverConf,
     started: AtomicBool,
+    // Keep the status sender wrapped in `Arc`, so we can (re)send it into the walreceiver manager loop,
+    // every time walreceiver is started.
+    // We do not allow more than one walreceiver manager loop, but can stop and start it periodically.
+    manager_status_sender: Arc<watch::Sender<Option<ConnectionManagerStatus>>>,
+    // Status sender needs a receiver end to send without errors, keep it to access the latest data sent.
+    manager_status_receiver: watch::Receiver<Option<ConnectionManagerStatus>>,
```

Comment:
Since we keep a reference to the sender, I don't see why we need to keep a reference to a receiver because there is [`watch::Sender::subscribe`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.subscribe). However, do we ever wait for this value to arrive?

I only see `Receiver::borrow` being used, meaning that we never really need any of the watch channel features. Did you consider just `tokio::sync::RwLock<Option<ConnectionManagerStatus>>`?

With the watch channel, we need to be mindful of updating the view the Receiver has, so I am now wondering if instead of `borrow` we should be using `borrow_and_update`? Well, looking at the implementation, there are no calls to [`Ref::has_changed`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Ref.html#method.has_changed) which makes me think of plain `tokio::sync::RwLock` again.

Reply:
If there's a real need for the watch features, could you point them out to me, I may have just blown past them.

Reply:
Yes, thank you for pointing out, replaced with RwLock.
I think, part of the reason I did not go with the solution was that it turns the method into `async` one, but looks like every place we care about the status (or could care later, as in HTTP responses) are async already.

Reply:
This is looking good.